### PR TITLE
Bugfix MTE-3550 Eliminating deprecated commands from Mergify

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3550)

## :bulb: Description
`merge` commands under `pull_request_rules` will be removed on Oct 2024. We should use the [Queue Rules](https://docs.mergify.com/configuration/file-format/#queue-rules). I determined that `merge_method` Queue Rule should be what we need.

I have not tested this change. Let me find out how to do so.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: .mergify.yml
Comment (clarmso) [inline .mergify.yml:3]: Mergify's IDE also told me that `condition` is invalid or deprecated.

**Comment by mattreaganmozilla** [comment]
Thanks @clarmso for investigating this.

**Comment by clarmso** [comment]
I've no idea why Mergify's editor tells me the config is valid, but the check still gives me the deprecation warning. Here's a PR opened from Mergify's editor: https://github.com/mozilla-mobile/firefox-ios/pull/22228

**Comment by clarmso** [comment]
Removing "Do not merge" label and marking the PR as ready for review.

I have toggled the labels "Needs Landing (Squash)", "Needs Landing" and no label for a (dummy)PR. The config editor determined that the config is valid and tells me the PR would go to the rebase vs squash queues.
Example: https://dashboard.mergify.com/config-editor?pullRequestNumber=22297

The "Action Required" section is still shown on the checks but I don't think the check is on this change. How about we merge this PR and see what happens @mattreaganmozilla ?

**Comment by mattreaganmozilla** [comment]
@clarmso Thanks for the heads-up 🙏 ; I'll defer to you and @isabelrios on merging but changes LGTM.

**Review by isabelrios — APPROVED** [review]
LGTM

**Pull request merged**
